prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -56,39 +115,424 @@ def _retrieve_redfish_sensor_data(self) -> List[Any]:
         Returns:
             sensors: List of dicts with details for each sensor
         """
+        sensors: List[Any] = []
+
+        logger.info("Getting redfish sensor info...")
+        sensors = redfish_utilities.get_sensors(self.redfish_obj)
+        return sensors
+
+    def _verify_redfish_call(self, redfish_obj: HttpClient, uri: str) -> Optional[Dict[str, Any]]:
+        """Return REST response for GET request API call on the URI.
+
+        Returns None if URI isn't present.
+        """
+        resp: RestResponse = redfish_obj.get(uri)
+        if resp.status == 200:
+            return resp.dict
```

Comment:
Is this possible return empty dictionary? What doesn't that mean if it happen?

Reply:
Yes, there could be a case where the URI is reachable (status is 200) but there's no elements present in it (empty dict). In my testing I haven't seen this case but it's good to also plan around this case.